packages/babel-helper-create-regexp-features-plugin/src/util.ts

```diff
@@ -3,22 +3,30 @@ import { FEATURES, hasFeature } from "./features";
 
 type RegexpuOptions = {
   unicodeFlag: "transform" | false;
+  unicodeSetsFlag: "transform" | "parse" | false;
   dotAllFlag: "transform" | false;
   unicodePropertyEscapes: "transform" | false;
   namedGroups: "transform" | false;
   onNamedGroup: (name: string, index: number) => void;
 };
 
+type Stable = 0;
+
 export function generateRegexpuOptions(toTransform: number): RegexpuOptions {
-  const feat = (name: keyof typeof FEATURES) => {
-    return hasFeature(toTransform, FEATURES[name]) ? "transform" : false;
+  const feat = <IsStable extends 0 | 1>(
+    name: keyof typeof FEATURES,
+    ok: "transform" | (IsStable extends Stable ? never : "parse") = "transform",
+  ) => {
+    return hasFeature(toTransform, FEATURES[name]) ? ok : false;
   };
 
   return {
-    unicodeFlag: feat("unicodeFlag"),
-    dotAllFlag: feat("dotAllFlag"),
-    unicodePropertyEscapes: feat("unicodePropertyEscape"),
-    namedGroups: feat("namedCaptureGroups"),
+    unicodeFlag: feat<Stable>("unicodeFlag"),
```

Comment:
nit: Since most flags are stable, we can use `Experimental = 1` in type notations.